app/process/barriers/barriers.lua

```diff
@@ -0,0 +1,123 @@
+-- Goal:
+-- ======
+-- Data on things that would prevent a bike network like water, highways, airports
+
+-- TODOs:
+-- ======
+-- LUA https://github.com/FixMyBerlin/osm-scripts/blob/main/utils/poiBarriers/downloadPoiBarriers.ts
+-- Remove small lakes. So closed ways that are of a certain area (LUA area function should work)
+-- Cleanup tags and add custom category for easy visualization
+
+package.path = package.path .. ";/app/process/helper/?.lua;/app/process/shared/?.lua"
+require("Set")
+require("FilterTags")
+require("Metadata")
+require("HighwayClasses")
+
+local lineBarriers = osm2pgsql.define_table({
+  name = 'lineBarriers',
+  ids = { type = 'any', id_column = 'osm_id', type_column = 'osm_type' },
+  columns = {
+    { column = 'tags', type = 'jsonb' },
+    { column = 'meta', type = 'jsonb' },
+    { column = 'geom', type = 'linestring' },
+  }
+})
+
+local areaBarriers = osm2pgsql.define_table({
+  name = 'areaBarriers',
+  ids = { type = 'any', id_column = 'osm_id', type_column = 'osm_type' },
+  columns = {
+    { column = 'tags', type = 'jsonb' },
+    { column = 'meta', type = 'jsonb' },
+    { column = 'geom', type = 'multipolygon' },
+  }
+})
+
+local excludedBarriers = osm2pgsql.define_table({
+  name = 'excludedBarriers',
+  ids = { type = 'any', id_column = 'osm_id', type_column = 'osm_type' },
+  columns = {
+    { column = 'tags', type = 'jsonb' },
+    { column = 'meta', type = 'jsonb' },
+    { column = 'geom', type = 'multipolygon' },
+  }
+})
+
+
+function isAreaBarrier(object)
+  local tags = object.tags
+  if object.type =='way' and not object.is_closed then
+    --only process ways iff closed
+    return false
+  end
+  local isBarrier = false
+
+  if tags.natural == 'water' then
+    local area = object:as_multipolygon():transform(3857):area()
+    tags.area = area
+    -- TODO: IMO it would be better to have this check on the single polygons of a multipolygon
+    if area > 100000 then
+      isBarrier = true
+    elseif object.type == 'way' then
+      isBarrier = isBarrier or object:as_linestring():transform(3857):length() > 1000
```

Comment:
Das sollten wir IMO nicht tun, weil wir damit Wege zerstückeln könnten. Szenario: Weg ist in drei Stücke zerteilt, das mittlere ist < 1000, dann würde es jetzt gelöscht. (Richtig?)

Reply:
IMO shouldn't be a problem because we check if the way is closed before and thus it should be one continuous geometry. But apart from that I think we need some evaluation on that condition, because it's kind of surpassing the area check. It's still important to fetch all canals and other narrow water ways. Maybe some kind of ratio between area and circumference could help.